install.sh

```diff
@@ -0,0 +1,176 @@
+}
+
+initDestination() {
+	if [ -n "$BINDIR" ]; then
+		if [ ! -d "$BINDIR" ]; then
+			fail "$BINDIR "'($BINDIR)'" folder not found. Please create it before continue."
+		fi
+		LBINDIR="$BINDIR"
+	else
+		if [ ! -d "$DEFAULT_BINDIR" ]; then
+		mkdir "$DEFAULT_BINDIR"
+		fi
+		LBINDIR="$DEFAULT_BINDIR"
+	fi
+	echo "Installing in $LBINDIR"
+}
+
+initArch() {
+	ARCH=$(uname -m)
+	case $ARCH in
+		armv5*) ARCH="armv5";;
+		armv6*) ARCH="armv6";;
+		armv7*) ARCH="ARM";;
+		aarch64) ARCH="ARM64";;
+		x86) ARCH="32bit";;
+		x86_64) ARCH="64bit";;
+		i686) ARCH="32bit";;
+		i386) ARCH="32bit";;
+	esac
+	echo "ARCH=$ARCH"
+}
+
+initOS() {
+	OS=$(uname -s)
+	case "$OS" in
+		Linux*) OS='Linux' ;;
+		Darwin*) OS='macOS' ;;
+		MINGW*) OS='Windows';;
+		MSYS*) OS='Windows';;
+	esac
+	echo "OS=$OS"
+}
+
+initDownloadTool() {
+	if type "curl" > /dev/null; then
+		DOWNLOAD_TOOL="curl"
+	elif type "wget" > /dev/null; then
+		DOWNLOAD_TOOL="wget"
+	else
+		fail "You need curl or wget as download tool. Please install it first before continue"
+	fi
+	echo "Using $DOWNLOAD_TOOL as download tool"
+}
+
+get() {
+	local url="$2"
+	local body
+	local httpStatusCode
+	echo "Getting $url"
+	if [ "$DOWNLOAD_TOOL" = "curl" ]; then
+		httpResponse=$(curl -sL --write-out HTTPSTATUS:%{http_code} "$url")
+		httpStatusCode=$(echo $httpResponse | tr -d '\n' | sed -e 's/.*HTTPSTATUS://')
+		body=$(echo "$httpResponse" | sed -e 's/HTTPSTATUS\:.*//g')
+	elif [ "$DOWNLOAD_TOOL" = "wget" ]; then
+		tmpFile=$(mktemp)
+		body=$(wget --server-response --content-on-error -q -O - "$url" 2> $tmpFile || true)
+		httpStatusCode=$(cat $tmpFile | awk '/^  HTTP/{print $2}')
+	fi
+	if [ "$httpStatusCode" != 200 ]; then
+		echo "Request fail with http status code $httpStatusCode"
+		fail "Body: $body"
+	fi
+	eval "$1='$body'"
+}
+
+getFile() {
+	local url="$1"
+	local filePath="$2"
+	if [ "$DOWNLOAD_TOOL" = "curl" ]; then
+		httpStatusCode=$(curl -s -w '%{http_code}' -L "$url" -o "$filePath")
+	elif [ "$DOWNLOAD_TOOL" = "wget" ]; then
+		body=$(wget --server-response --content-on-error -q -O "$filePath" "$url")
+		httpStatusCode=$(cat $tmpFile | awk '/^  HTTP/{print $2}')
+	fi
+	echo "$httpStatusCode"
+}
+
+downloadFile() {
+	get TAG_JSON https://api.github.com/repos/arduino/arduino-cli/releases/latest
+	TAG=$(echo $TAG_JSON | python -c 'import json,sys;obj=json.load(sys.stdin);sys.stdout.write(obj["tag_name"])')
+	echo "TAG=$TAG"
+	#  arduino-cli_0.4.0-rc1_Linux_64bit.tar.gz
+	CLI_DIST="arduino-cli_${TAG}_${OS}_${ARCH}.tar.gz"
+	echo "CLI_DIST=$CLI_DIST"
+	DOWNLOAD_URL="https://downloads.arduino.cc/arduino-cli/$CLI_DIST"
+	CLI_TMP_FILE="/tmp/$CLI_DIST"
+	echo "Downloading $DOWNLOAD_URL"
+	httpStatusCode=$(getFile "$DOWNLOAD_URL" "$CLI_TMP_FILE")
+	if [ "$httpStatusCode" -ne 200 ]; then
+		echo "Did not find a release for your system: $OS $ARCH"
+		echo "Trying to find a release on the github api."
+		LATEST_RELEASE_URL="https://api.github.com/repos/Masterminds/$PROJECT_NAME/releases/tags/$TAG"
```

Comment:
There's a leftover from the Mastermind project:
```suggestion
		LATEST_RELEASE_URL="https://api.github.com/repos/arduino/$PROJECT_NAME/releases/tags/$TAG"
```